### PR TITLE
Add read_dataframe api.

### PR DESCRIPTION
**Pull request by shifucun**

### CODE REVIEW

File: datacommons/datacommons.py [hunk 1]
```diff
@@ -400,9 +400,34 @@ def get_observations(self,
 
   # -------------------------- CACHING FUNCTIONS --------------------------
 
+  def read_dataframe(self, file_name):
+    """Read pandas dataframe from caching.
```
Comment (pradh) [inline datacommons/datacommons.py]: Read a previously saved pandas dataframe.

Comment (pradh) [inline datacommons/datacommons.py]: cache

File: datacommons/datacommons.py [hunk 2]
```diff
@@ -400,9 +400,34 @@ def get_observations(self,
...
+
+      Use can only read previously saved data file with the same authentication
```
Comment (pradh) [inline datacommons/datacommons.py]: nit: User

File: datacommons/examples/simple_query.py [hunk 3]
```diff
@@ -45,7 +45,10 @@ def main():
   with pd.option_context('display.width', 400, 'display.max_rows', 100):
     print(df)
 
-  dc.save_dataframe(df, 'simple_query_result')
+  dc.save_dataframe(df, 'test_df')
+  saved_pd = dc.read_dataframe('test_df')
+  with pd.option_context('display.width', 400, 'display.max_rows', 100):
+    print(saved_pd)
```
Comment (pradh) [inline datacommons/examples/simple_query.py]: Can you compare the original and saved/restored dataframes?

File: datacommons/datacommons.py (hunk 1 above)
Comment (shifucun) [reply datacommons/datacommons.py]: Updated

File: datacommons/examples/simple_query.py (hunk 3 above)
Comment (shifucun) [reply datacommons/examples/simple_query.py]: Added comparison

**Pull request merged**